File: frontend/src/main/java/com/innertrack/model/ContactRequest.java
package com.innertrack.model;

import java.time.LocalDateTime;

public class ContactRequest {
    private int id;
    private int clientId;
    private int therapistId;
    private String message;
    private String status; // PENDING | ACCEPTED | DECLINED
    private LocalDateTime createdAt;
    private LocalDateTime respondedAt;

    // Transient
    private String clientName;

    public ContactRequest() {}

    public ContactRequest(int clientId, int therapistId, String message) {
        this.clientId = clientId;
        this.therapistId = therapistId;
        this.message = message;
        this.status = "PENDING";
    }

    public int getId()                           { return id; }
    public void setId(int id)                    { this.id = id; }
    public int getClientId()                     { return clientId; }
    public void setClientId(int v)               { this.clientId = v; }
    public int getTherapistId()                  { return therapistId; }
    public void setTherapistId(int v)            { this.therapistId = v; }
    public String getMessage()                   { return message; }
    public void setMessage(String v)             { this.message = v; }
    public String getStatus()                    { return status; }
    public void setStatus(String status)         { this.status = status; }
    public LocalDateTime getCreatedAt()          { return createdAt; }
    public void setCreatedAt(LocalDateTime v)    { this.createdAt = v; }
    public LocalDateTime getRespondedAt()        { return respondedAt; }
    public void setRespondedAt(LocalDateTime v)  { this.respondedAt = v; }
    public String getClientName()                { return clientName; }
    public void setClientName(String v)          { this.clientName = v; }
}

File: backend/src/Entity/ContactRequest.php
<?php

namespace App\Entity;

use Doctrine\ORM\Mapping as ORM;

#[ORM\Entity]
#[ORM\Table(name: 'contact_request')]
class ContactRequest
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column]
    private ?int $id = null;

    #[ORM\ManyToOne(targetEntity: User::class)]
    #[ORM\JoinColumn(name: 'client_id', referencedColumnName: 'id', nullable: false)]
    private User $client;

    #[ORM\ManyToOne(targetEntity: User::class)]
    #[ORM\JoinColumn(name: 'therapist_id', referencedColumnName: 'id', nullable: false)]
    private User $therapist;

    #[ORM\Column(type: 'text', nullable: true)]
    private ?string $message = null;

    #[ORM\Column(length: 20, options: ['default' => 'PENDING'])]
    private string $status = 'PENDING';

    #[ORM\Column(name: 'created_at', type: 'datetime')]
    private \DateTimeInterface $createdAt;

    #[ORM\Column(name: 'responded_at', type: 'datetime', nullable: true)]
    private ?\DateTimeInterface $respondedAt = null;

    public function __construct()
    {
        $this->createdAt = new \DateTime();
    }

    public function getId(): ?int { return $this->id; }
    public function getClient(): User { return $this->client; }
    public function setClient(User $client): self { $this->client = $client; return $this; }
    public function getTherapist(): User { return $this->therapist; }
    public function setTherapist(User $therapist): self { $this->therapist = $therapist; return $this; }
    public function getMessage(): ?string { return $this->message; }
    public function setMessage(?string $message): self { $this->message = $message; return $this; }
    public function getStatus(): string { return $this->status; }
    public function setStatus(string $status): self { $this->status = $status; return $this; }
    public function getCreatedAt(): \DateTimeInterface { return $this->createdAt; }
    public function getRespondedAt(): ?\DateTimeInterface { return $this->respondedAt; }
    public function setRespondedAt(?\DateTimeInterface $respondedAt): self { $this->respondedAt = $respondedAt; return $this; }

    public function isPending(): bool { return $this->status === 'PENDING'; }
}
